### ISSUE TITLE
Jasmine example shows no cli output

### ISSUE THREAD

**Issue by davemckenna01** [jasmine/jasmine-npm#88]
I followed everything here in the README of this repo, but I get no output on the cli when I run ```jasmine```. I must be missing something obvious.

```which jasmine``` shows ```./node_modules/.bin/jasmine``` which is in my project dir.

I run with explicit locations and get the same result: ```JASMINE_CONFIG_PATH=spec/support/jasmine.json jasmine spec/jasmine_examples/PlayerSpec.js```

By no output I mean:
`prompt>$jasmine`
`prompt>$`

Any ideas?

**Comment by davemckenna01** [comment]
On further investigation, it looks like I need to specify a reporter. Well an exit code reporter seems to be default and bash `$?` tells me what the exit code is.

Might it be worth mentioning this in the README/docs? I feel like people might expect to see command line output when they're getting started running the example tests.

**Comment by davemckenna01** [comment]
Does jasmine ship with a terminal reporter?

**Comment by kaicataldo** [comment]
Was seeing results until the latest version. Now doesn't report anything. Rolling back to 2.4.1 reports results as expected. 

Looking around and not seeing how I add a custom reporter either as a CLI flag or in the `jasmine.json` file. Any help would be appreciated!

**Comment by amavisca** [comment]
Hmm, not exactly sure why this is happening off the top of my head. Looks like a `provideFallbackReporter` was added to jasmine-core's env interface which we just call through to. It might be good to look there.

**Comment by flore77** [comment]
I have also a similar problem and I think this is because now in the 2.5.0 version, the [configureDefaultReporter](https://github.com/jasmine/jasmine-npm/blob/164b799c6e531f29b23722cf6975dfe1400ac7b6/lib/jasmine.js#L144) is always called and [defaultReporterAdded](https://github.com/jasmine/jasmine-npm/blob/164b799c6e531f29b23722cf6975dfe1400ac7b6/lib/jasmine.js#L163) will always be true and will enter this branch exiting the process.

The `defaultReporterAdded` prop should [not be set always to true](https://github.com/jasmine/jasmine-npm/blob/164b799c6e531f29b23722cf6975dfe1400ac7b6/lib/jasmine.js#L68), because now only jasmine-core knows if any reporter has been added and if the default reporter should be used, https://github.com/jasmine/jasmine/blob/c7cc3b4a29e92bf59fe56d5a8982e87b58c5c1d7/src/core/ReportDispatcher.js#L30.

**Comment by agirorn** [comment]
Is there a workaround for this bug until a new jasmine is published?

**Comment by agirorn** [comment]
This is still broken for everyone that has "jasmine": "^2.4.1" and "jasmine": "^2" in there package.json, They will get 2.5.0 on next fresh npm install and experience this issue. Are you going to publish 2.5.1 with this fix ?

**Comment by slackersoft** [comment]
Fix should be released in 2.5.1

**Comment by bcaudan** [comment]
As said in #90, it still not working with 2.5.1.

**Comment by Liero** [comment]
yes, still does not work in 2.5.1. Please reopen, otherwise new issues will be reported

**Comment by flore77** [comment]
Hey,

The fix that I applied for this issue, unfortunately, does not work for the defaultReporter. It is working for third-parties reporter, it was the same problem like in #43.

How @bcaudan pointed out in #90 the problem is:

> It seems that the console reporter is not used because it is only provided as a fallback reporter. However, there is always another reporter provided, the exitCodeReporter.

**Comment by nashwaan** [comment]
Jasmine cli does not hang in v2.5.1 as it used to be in v2.5.0, but still no jasmine report output in the command line in v2.5.1.

@slackersoft Could you explain how to get the console reporter to work as it used to be in v2.4.1?

If this is dropped in favor of 'exitCodeReported', the how to get the 'consoleReporter' working again???
Sample code, please?

**Comment by em0ney** [comment]
I'm moving back to version 2.4.1 until this is resolved

```
"jasmine": "~2.4.1",
```

**Comment by unconed** [comment]
For a quick solution, you can drop in a `spec/helpers/consolereporter.js` to get output back:

```
var util = require('util');
var options = {
    showColors: true,
    print: function() {
        process.stdout.write(util.format.apply(this, arguments));
    }
};

jasmine.getEnv().addReporter(new jasmine.ConsoleReporter(options));
```

Moral of the story: if you have pluggable output modules, be nice and warn if there's no output hooked up at all, before people waste an hour or two each debugging mysteries.

**Comment by peterhal** [comment]
I just got bit by this. This issue is still breaking everyone with

jasmine: "^2.4.1"

In their package.json.

**Comment by nashwaan** [comment]
Thanks alot @unconed. Your solution works!
I don't know where you got this, but I think this need to be mentioned in the docs by the jasmine-npm team. People should not be expected to find this solution in the issues section of the github project.

Now I get this output:
```bash
ConsoleReporter is deprecated and will be removed in a future version. Started
....................
20 specs, 0 failures
Finished in 0 seconds
```
But notice the first line says ConsoleReporter `will be removed`. What are the alternatives in order to avoid this functionality to be broken in the future?

**Comment by slackersoft** [comment]
It sounds like this wasn't actually fixed as I thought it was with the 2.5.1 release. Hopefully, I should have some time today to take a look at it and see what the problem is. 

The intent of the reporter changes was to remove the need for a custom reporter to have to tell Jasmine about full success/fail of the test run and just print the output that it wants.

**Comment by m-a-r-c-e-l-i-n-o** [comment]
@slackersoft and people of this thread. Here's some light at the end of the tunnel: https://github.com/jasmine/jasmine/pull/1208
My jasmine knowledge is fairly limited, but I was itching to get this moving along, hopefully it helps.

**Comment by slackersoft** [comment]
Alrighty, I think I know what happened here and what the problem is, and I have a workaround for now and some options I'm mulling over for real fixes.

First, why this is happening:

1. Jasmine-npm provides _its_ `ConsoleReporter` to jasmine-core as a fallback reporter to be used if no other reporters are added
  1. This should then allow users to specify custom reporters directly to the jasmine-core environment without having to use the `addReporter` function in jasmine-npm
1. Just before starting execution, jasmine-npm add its `ExitCodeReporter` to the jasmine-core environment
1. When jasmine-core starts reporting, it sees that it has a reporter (ExitCode) and doesn't use the fallback reporter provided.

If you want to keep use the most current `ConsoleReporter` you should use the one provided in this package. It should be exported on the main `jasmine` object as `ConsoleReporter`. Something like this:

```javascript
var jasmineNpm = require('jasmine');
jasmine.getEnv().addReporter(new jasmineNpm.ConsoleReporter(...));
```

The two primary options I'm thinking of for long-term solutions are:

1. Plum a way for the `ExitCodeReporter` to be added to jasmine-core such that it doesn't count against the fallback reporter
1. Allow libraries to check if a jasmine-core environment has any reporters registered, so jasmine-npm can only add `ConsoleReporter` if there aren't any other reporters.

I would love to hear any thoughts on these solutions or any suggestions for another way to solve this.

Thanks for your patience, and sorry for the premature closing earlier.

**Comment by m-a-r-c-e-l-i-n-o** [comment]
In your first solution, would the `jasmine-core` be aware of the actual `ExitCodeReporter` reporter? Or do you plan to pass in an "offset" flag (i.e. `addReporter(ExitCodeReporter, 'dont-count-me')` -- essentially telling it not to count "whatever" this reporter is)? If the latter, I would say it sounds like a winner.

The second solution sounds okay to me, in that it's non intrusive, but could be misleading if the reported count is not requested immediately before execution. So maybe it would be worth it to make this very clear in the docs, or designate a flag in the `addReporter` method (i.e. `addReporter(ConsoleReporter, 'final')`), which causes `jasmine-core` to throw an error if any other reporters are added after this "final" reporter is added.

**Comment by slackersoft** [comment]
Due to the direction of the dependency chain, it would be hard to sensibly allow jasmine-core to know specifically about the `ExitCodeReporter` from jasmine-npm, so it would likely be some kind of flag to tell jasmine-core to ignore this reporter when checking if any have been registered before adding the fallback.

The second option probably wouldn't include any way to fail if a reporter was added after another given reporter, so it would just be up to jasmine-npm to add `ConsoleReporter` just before starting the run if no other reporters are registered.

I'm leaning toward the second option because it is a bit less involved, but it also seems to remove the need for a fallback reporter at all...

**Comment by th317erd** [comment]
It seems to me like this is kind of backwards to begin with. Shouldn't jasmine-core contain and use the ExitCodeReporter by default if no other reporter exists, and jasmine-npm can set the ConsoleReporter?

**Comment by slackersoft** [comment]
Jasmine-npm is the only Jasmine runner that I know of that it makes sense to have a direct exit code produced from the run. Most other runners for Jasmine are intended for use in the browser where having an exit code is nonsensical.

**Comment by th317erd** [comment]
So instead of setting the ConsoleReporter as a FALLBACK in jasmine-npm why don't you just set both reporters? 

```javascript
addReporter(new ConsoleReporter(...))
addReporter(new ExitCodeReporter(...))
```

**Comment by m-a-r-c-e-l-i-n-o** [comment]
@th317erd Causing conflicts with other reporters would be one reason not to do that.

**Comment by slackersoft** [comment]
@th317erd because if jasmine-npm always registers its `ConsoleReporter` it is impossible for users of jasmine-npm to not include it when they want to customize the output. This is sort of the root cause of the genesis of this problem.

**Comment by bcaudan** [comment]
I proposed another way to solve this with #92.

It seems a bit odd to handle exit code with a reporter but it was working for now. With a way to handle an env execution callback in jasmine, then ExitCodeReporter could be replaced by a jasmine-npm default onComplete callback. This callback could be overriden by the existing onComplete option.

Then jasmine-npm would only provide the console reporter as a fallback reporter.

maybe I am missing something, what do you think of this strategy?

**Comment by m-a-r-c-e-l-i-n-o** [comment]
@slackersoft 
Interesting approach, really gets at the heart of the problem -- giving the user choice :)
Thank you.

**Comment by nashwaan** [comment]
Thanks @slackersoft 
The problem is now solved in v2.5.2
